### PR TITLE
[NEW] URL mappings for jax-rs/jersey

### PR DESCRIPTION
**Pull request by monu-k2io**
Feature for getting url mappings for jax-rs/jersey

### CODE REVIEW

File: instrumentation-security/jersey/src/main/java/com/nr/instrumentation/security/jersey/JerseyHelper.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import java.util.Set;
+
+public class JerseyHelper {
+    private static final String EMPTY = "";
+    private static final String WILDCARD = "*";
+    private static final String SEPARATOR = "/";
+    public static void gatherUrlMappings(ResourceModel resourceModel) {
+        try {
+            List<Resource> resources = resourceModel.getResources();
+            extractMappingsFromResources(resources, EMPTY);
```
Comment (lovesh-ap) [inline instrumentation-security/jersey/src/main/java/com/nr/instrumentation/security/jersey/JerseyHelper.java:20]: Null check for resources is missing

File: instrumentation-security/jersey/src/main/java/com/nr/instrumentation/security/jersey/JerseyHelper.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+                if(resource.getAllMethods().size() > 0){
+                    for (ResourceMethod method: resource.getAllMethods()){
+                        String httpMethod = method.getHttpMethod();
+                        if(httpMethod != null){
+                            addURLMappings(url, httpMethod, resource.getHandlerClasses());
+                        } else {
+                            // httpMethod is null in case when method represents a sub-resource locator.
+                            String modifiedUrl = url + SEPARATOR + WILDCARD;
+                            addURLMappings(WILDCARD, modifiedUrl, resource.getHandlerClasses());
```
Comment (lovesh-ap) [inline instrumentation-security/jersey/src/main/java/com/nr/instrumentation/security/jersey/JerseyHelper.java:45]: the call is incorrect, modifiedURL and WILDCARD has switched the calling sequence

File: instrumentation-security/jersey/src/main/java/com/nr/instrumentation/security/jersey/JerseyHelper.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+                        }
+                    }
+                } else if((resource.getChildResources().size() == 0)){
+                    addURLMappings(WILDCARD, url, resource.getHandlerClasses());
```
Comment (lovesh-ap) [inline instrumentation-security/jersey/src/main/java/com/nr/instrumentation/security/jersey/JerseyHelper.java:49]: same incorrect calling sequence

**Review by lovesh-ap — APPROVED** [review]

File: instrumentation-security/jersey/src/main/java/com/nr/instrumentation/security/jersey/JerseyHelper.java (hunk 1 above)
Comment (IshikaDawda) [reply instrumentation-security/jersey/src/main/java/com/nr/instrumentation/security/jersey/JerseyHelper.java]: Added null check in commit [f857a83](https://github.com/newrelic/csec-java-agent/pull/56/commits/f857a833a0698b3357b6fdf8fc1d9cb16fa8b72a)

File: instrumentation-security/jersey/src/main/java/com/nr/instrumentation/security/jersey/JerseyHelper.java (hunk 2 above)
Comment (IshikaDawda) [reply instrumentation-security/jersey/src/main/java/com/nr/instrumentation/security/jersey/JerseyHelper.java]: Corrected the calling sequence in [f857a83](https://github.com/newrelic/csec-java-agent/pull/56/commits/f857a833a0698b3357b6fdf8fc1d9cb16fa8b72a)

**Pull request merged**